src/main/java/com/epam/aidial/core/log/GfLogStore.java

```diff
@@ -95,7 +96,27 @@ private void append(ProxyContext context, LogEntry entry) {
         append(entry, "\",\"body\":\"", false);
         append(entry, context.getResponseBody());
 
-        append(entry, "\"}}", false);
+        TokenUsage tokenUsage = context.getTokenUsage();
+        if (tokenUsage != null) {
+            append(entry, "\"},\"tokenUsage\":{", false);
+            append(entry, "\"completion_tokens\":", false);
+            append(entry, Long.toString(tokenUsage.getCompletionTokens()), true);
+            append(entry, ",\"prompt_tokens\":", false);
+            append(entry, Long.toString(tokenUsage.getPromptTokens()), true);
+            append(entry, ",\"total_tokens\":", false);
+            append(entry, Long.toString(tokenUsage.getTotalTokens()), true);
+            if (tokenUsage.getCost() != null) {
+                append(entry, ",\"cost\":", false);
+                append(entry, tokenUsage.getCost().toString(), true);
+            }
+            if (tokenUsage.getAggCost() != null) {
```

Comment:
maybe totalCost is better?

Reply:
We agreement with @avavilau to name this kind of costs as aggregated. If you have objections feel free to discuss that with the team.